core/src/main/java/io/grpc/internal/DelayedClientTransport.java

```diff
@@ -335,14 +339,17 @@ void startBackoff(final Status status) {
             it.remove();
           }
         }
-        streamCreationExecutor.execute(new Runnable() {
+
+        class FailTheFailFastPendingStreams implements Runnable {
```

Comment:
it was used in the test, I'm not sure if there is a more convenient way in the test to run `startBackoff` and `TransportSet#shutdown` in the mean time.